test/e2e/mongodbtests/mongodbtests.go

```diff
@@ -289,6 +289,19 @@ func BasicFunctionality(mdb *mdbv1.MongoDBCommunity) func(*testing.T) {
 	}
 }
 
+// ServiceWithNameExists checks whether a service with the name serviceName exists
+func ServiceWithNameExists(mdb *mdbv1.MongoDBCommunity, serviceName string) func(t *testing.T) {
```

Comment:
[nit] we are introducing a dependency on  `*mdbv1.MongoDBCommunity` when in realsity all we actually care about is the namespace. This signature could be `(namespace, serviceName string)` or even `(nsName types.NamespacedName)`

Reply:
This should happen more often in functions, I think `mongodbtests.go` has quite a few such examples -- e.g `StatefulSetHasOwnerReference` ...